src/components/exchange/AvailableNetworksv2.tsx

```diff
@@ -0,0 +1,233 @@
+          }),
+          onSelectCurrency: updateInputCurrency,
+        },
+        screen: Routes.CURRENCY_SELECT_SCREEN,
+      });
+    },
+    [asset, goBack, navigate, networks, updateInputCurrency]
+  );
+
+  const handlePressContextMenu = useCallback(
+    ({ nativeEvent: { actionKey: network } }) => {
+      const chosenNetwork = network;
+      convertAssetAndNavigate(chosenNetwork);
+    },
+    [convertAssetAndNavigate]
+  );
+
+  const handlePressButton = useCallback(() => {
+    const chosenNetwork = availableNetworks[0];
+    convertAssetAndNavigate(chosenNetwork);
+  }, [availableNetworks, convertAssetAndNavigate]);
+
+  const networkMenuItems = useMemo(() => {
+    return Object.values(networkInfo)
+      .filter(
+        ({ exchange_enabled, value }) =>
+          exchange_enabled &&
+          value !== Network.mainnet &&
+          !!networks[ethereumUtils.getChainIdFromNetwork(value)]
+      )
+      .map(netInfo => ({
+        actionKey: netInfo.value,
+        actionTitle: netInfo.name,
+        icon: {
+          iconType: 'ASSET',
+          iconValue: `${
+            netInfo.layer2 ? `${netInfo.value}BadgeNoShadow` : 'ethereumBadge'
+          }`,
+        },
+      }));
+  }, [networks]);
+
+  const MenuWrapper =
+    availableNetworks.length > 1 ? ContextMenuButton : React.Fragment;
+
+  return (
+    <>
+      <MenuWrapper
+        // @ts-expect-error overloaded props ContextMenuButton
+        menuConfig={{ menuItems: networkMenuItems, menuTitle: '' }}
+        isMenuPrimaryAction
+        onPressMenuItem={handlePressContextMenu}
+        useActionSheetFallback={false}
+      >
+        <ButtonPressAnimation
+          scaleTo={0.96}
+          onPress={availableNetworks.length === 1 ? handlePressButton : NOOP}
+        >
+          <Box
+            borderRadius={99}
+            padding="8px"
+            paddingHorizontal={{ custom: 16 }}
+            marginHorizontal={{ custom: marginHorizontal }}
+            justifyContent="center"
+          >
+            <RadialGradient
+              {...radialGradientProps}
+              // @ts-expect-error overloaded props RadialGradient
+              borderRadius={99}
+              radius={600}
+            />
+            <Inline alignVertical="center" alignHorizontal="justify">
+              <Inline alignVertical="center">
+                <Box style={{ flexDirection: 'row' }}>
+                  {availableNetworks?.map((network, index) => {
+                    return (
+                      <Box
+                        background="body"
+                        key={`availbleNetwork-${network}`}
+                        marginLeft={{ custom: -6 }}
+                        style={{
+                          backgroundColor: colors.transparent,
+                          zIndex: availableNetworks?.length - index,
+                          borderRadius: 30,
+                        }}
+                        width={{ custom: 22 }}
```

Comment:
i don't think we need `width` and the `marginLeft` should be -4 https://www.figma.com/file/SFfYzRoHlapEecS0LTLwhQ/Swap-Aggregator-V2?node-id=898%3A56903